File: 상한규/프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.js
function solution(participant, completion) {
    var answer = '';
    
    participant.sort();
    completion.sort();
    
    for(let i = 0; i < completion.length; i++) {
        if(participant[i] != completion[i]) {
            answer = participant[i];
            break;
        }
    }
    
    if(answer == '') {
        answer = participant[participant.length - 1];
    }
    
    return answer;
}

File: 방승윤/프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.java
import java.util.HashMap;

class Solution {
    public String solution(String[] participant, String[] completion) {
        String answer = "";

        HashMap<String, Integer> hashMap = new HashMap<String, Integer>();

        for (String c : completion) {
            hashMap.put(c, hashMap.getOrDefault(c, 0) + 1);
        }

        for (String p : participant) {
            if (!hashMap.containsKey(p)) {
                return p;
            }

            if (hashMap.get(p) == 0) {
                return p;
            }

            hashMap.replace(p, hashMap.get(p) - 1);
        }

        return "";
    }
}